### ISSUE TITLE
[Bug]: Cannot create a new user in Ubuntu hirsute

### ISSUE THREAD

**Issue by WheezyE** [termux/proot-distro#159]
### Problem description

I'm trying to use proot-distro Ubuntu hirsute on Termux, but when I try to create a new user account, I encounter "lock file already used" errors:
```
root@localhost:~# useradd -m -r user
useradd: /etc/passwd.26979: lock file already used
useradd: cannot lock /etc/passwd; try again later.
root@localhost:~# useradd -m -r user
useradd: /etc/passwd.27016: lock file already used
useradd: cannot lock /etc/passwd; try again later.
root@localhost:~# useradd -m user
useradd: /etc/passwd.27017: lock file already used
useradd: cannot lock /etc/passwd; try again later.
root@localhost:~# adduser user
Adding user `user' ...
Adding new group `user' (1000) ...
groupadd: /etc/group.27041: lock file already used
groupadd: cannot lock /etc/group; try again later.
adduser: `/sbin/groupadd -g 1000 user' returned error code 10. Exiting.
root@localhost:~# adduser user5
Adding user `user5' ...
Adding new group `user5' (1000) ...
groupadd: /etc/group.27044: lock file already used
groupadd: cannot lock /etc/group; try again later.
adduser: `/sbin/groupadd -g 1000 user5' returned error code 10. Exiting.
root@localhost:~#
```

My eventual goal is to run wine in a user account on Termux.  I had a working version in the past at https://github.com/WheezyE/AnBox86/blob/main/AnBox86.sh  That version used proot-static, but testing here is without proot-static, just using vanilla proot-distro.

### What steps will reproduce the bug?

In a freshly-installed Termux (from F-Droid):
```
pkg install proot-distro git -y
linux32 proot-distro install ubuntu
proot-distro login --isolated ubuntu
```
In proot:
```
apt update -y
adduser user
```

I've also tried other user account names, logging out of proot and logging back in, trying it without --isoloated, trying it without `apt update -y` (all these attempts were with fresh Termux installations - uninstalled Termux and reinstalled).

### What is the expected behavior?

Before (in proot-distro Ubuntu-20.04) I was able to create a user accounts.

**Comment by ZhymabekRoman** [comment]
Yeah, same problem: 
![image](https://user-images.githubusercontent.com/61125068/134750332-b66f308a-3b3f-402e-9615-ad777422a0e5.png)

**Comment by xeffyr** [comment]
Maybe `proot` bug affecting some devices or , but definitely not an issue of `proot-distro`. I can't reproduce it...

![Screenshot_20210925-113214~2](https://user-images.githubusercontent.com/25881154/134765171-65ee909a-fd56-4c9d-9349-fe514ad9e35a.png)

**Comment by xeffyr** [comment]
Please don't use `linux32` to run `proot-distro`. This is not supported.

If you want 32 bit distribution, then cd into `$PREFIX/etc/proot-distro`, create a copy of plug-in (e.g. `cp debian.sh debian-32bit.sh`) and edit it to add `DISTRO_ARCH` variable with desired architecture.

**Comment by WheezyE** [comment]
Sorry if I'm not quite understanding this right.  And thank you for the tip about DISTRO_ARCH.  I'm still getting user account setup permission errors though:
In a fresh Termux install, I installed proot-distro, ran `cp $PREFIX/etc/proot-distro/ubuntu.sh $PREFIX/etc/proot-distro/ubuntu-32bit.sh` and then edited ubuntu-32bit.sh so that `DISTRO_ARCH="arm"` was at the top, then ran `proot-distro install ubuntu-32bit` which successfully installed the 32bit armhf packages.  When I log into proot though with `proot-distro login --isolated ubuntu-32bit` and run `arch` it reports `aarch64` and user account creations still have the same errors I reported before:

```
Adding new group `user' (1000) ...
groupadd: /etc/group.15184: lock file already used
groupadd: cannot lock /etc/group; try again later.
adduser: `/sbin/groupadd -g 1000 user' returned error code 10. Exiting
```

For more context, I think my use-case is similar to https://github.com/termux/proot-distro/issues/48#issuecomment-918483318 where I want a 32-bit system to run box86 (not qemu).  I understand that this might not be the intended use for proot-distro though.

My ultimate goal is actually to run a aarch64 environment anyway with armhf multiarch.  It's just odd that this was working previously in ubuntu-20.04.

**Comment by xeffyr** [comment]
You can ask @michalbednarski in https://github.com/termux/proot/issues about this error. Anyway, `proot-distro` is only distribution rootfs manager script and it doesn't emulate anything on its own. All runtime issues happen due to `proot` utility.

> When I log into proot though with proot-distro login --isolated ubuntu-32bit and run arch it reports aarch64

Better to check with `dpkg --print-architecture`, it will print the current architecture of installed packages. Yes, `arch` and `uname -m` report `aarch64`, but `dpkg` reports `armhf`.

> It's just odd that this was working previously in ubuntu-20.04.

Then the latest Ubuntu version use newer syscalls in glibc or somewhere else and `proot` really need a fix. There are chances that other distributions soon will be affected as well.

**Comment by ZhymabekRoman** [comment]
> Maybe `proot` bug affecting some devices or , but definitely not an issue of `proot-distro`. I can't reproduce it...

@xeffyr, you have the latest version of the proot?

**Comment by xeffyr** [comment]
The latest and `aarch64`.

**Comment by WheezyE** [comment]
Just wanted to update this thread with the solution: You were right that it was a bug in proot.  This has now been fixed by @michalbednarski in `proot 5.1.107-44`  https://github.com/termux/proot/issues/191#issuecomment-927328982
